testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java

```diff
@@ -937,32 +942,41 @@ SpanDataAssert assertClientSpan(
       String method,
       @Nullable Integer responseCode,
       @Nullable Integer resendCount) {
-    Set<AttributeKey<?>> httpClientAttributes = options.getHttpAttributes().apply(uri);
+    Set<AttributeKey<?>> httpClientAttributes =
+        getAttributeKeys(options.getHttpAttributes().apply(uri));
     return span.hasName(options.getExpectedClientSpanNameMapper().apply(uri, method))
         .hasKind(SpanKind.CLIENT)
         .hasAttributesSatisfying(
             attrs -> {
               // TODO: Move to test knob rather than always treating as optional
-              if (attrs.get(SemanticAttributes.NET_TRANSPORT) != null) {
+              if (SemconvStability.emitOldHttpSemconv()
+                  && attrs.get(SemanticAttributes.NET_TRANSPORT) != null) {
```

Comment:
for new semconv does it make sense to check `NETWORK_TRANSPORT`, and `NETWORK_TYPE`?

Reply:
I think `NETWORK_TRANSPORT` is never set. Added assertion for `NETWORK_TYPE`, should always be `ipv4`.

Reply:
> I think `NETWORK_TRANSPORT` is never set.

Grizzly and Netty set it. We could just check for null or `tcp`, it's a non-required attribute anyway.

Reply:
Ohh, you're right, it does prevent that.

> Or should assert that it is null?

Yeah, we could do that.